windows_hotfix resource: test whether a Windows HotFix is installed

mattray opened the pull request:
Adds the ability to check if WIndows hotfixes of the form "KB1234567" are installed or not.

Signed-off-by: Matt Ray <matthewhray@gmail.com>

mattray committed:
Add hotfix resource for Windows

Signed-off-by: Matt Ray <matthewhray@gmail.com>

adamleff commented on test/unit/resources/hotfix_test.rb:
```diff
@@ -0,0 +1,20 @@
+  it 'verify hotfix on windows' do
+    resource = MockLoader.new(:windows).load_resource('hotfix', 'KB4019215')
+    _(resource.installed?).must_equal true
+  end
```
Can we add a test for a hotfix that doesn't exist, along with the mock command output we should expect?

adamleff reviewed (changes requested):
Hi @mattray! Thanks for this, this is a great addition.

Any chance I can convince you to name the resource `windows_hotfix` instead so we don't have to worry about resource name confusion if we ever have a hotfix-like resource for something else?

Re: integration test, one way of doing this might be to have the chef recipe that sets up the test node query for any installed hotfixes and write out an installed hotfix ID to the filesystem, and then the integration test could read from that file. That way we don't have to worry about catering to different OS flavors and potential roll-ups eating individual IDs.

mattray committed:
Renamed hotfix to windows_hotfix

Added additional unit test checking for KB that is not present on a box

Signed-off-by: Matt Ray <matthewhray@gmail.com>

mattray committed:
Integration test to spot-check for hotfixes

Queries the Windows operating system via Powershell for a list of all
installed hotfixes and spot-checks every 10th one with the
windows_hotfix resource. Checking hundreds is time-consuming. Also
checks to ensure a non-installed hotfix is not present.

Signed-off-by: Matt Ray <matthewhray@gmail.com>

adamleff reviewed (approved):
This looks great, @mattray. Thank you!

arlimus reviewed (approved):
This is a great addition, thank you @mattray !!
